demo/DemoArticles/update.js

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env node
+/* eslint-disable require-jsdoc */
+
+const fs = require('fs')
+const request = require('request')
+const indexJSON = require('./articles.json')
+const ArticleRef = require('./ArticleRef.js').ArticleRef
+const ArticleRefSourceType = require('./ArticleRef.js').ArticleRefSourceType
+const flattenArrayOfArrays = require('../DemoUtilities.js').flattenArrayOfArrays
+
+const DATA_PATH = './data/'
+
+const articleRefs = flattenArrayOfArrays(
+  indexJSON.map(articleData => [
+    new ArticleRef(
+      articleData.lang,
+      articleData.title,
+      articleData.revision,
+      ArticleRefSourceType.mobileView
```

Comment:
For future refactor consideration: storing the backend type is needlessly a property of ArticleRef. The client of ArticleRef can just supply this to a `ArticleRef.url(ArticleRefSourceType)` method or (even better) supply the ArticleRef to `ArticleRefSourceType.url(ArticleRef)`.